crates/biome_js_analyze/src/analyzers/style/use_template.rs

```diff
@@ -150,51 +157,94 @@ fn can_be_template_literal(node: &JsBinaryExpression) -> Option<bool> {
 fn template_expression_from_binary_expression(
     node: &JsBinaryExpression,
 ) -> Option<JsTemplateExpression> {
-    // While `template_elements` is empty, we keep track of the last left node.
-    // Once we see the first string/template literal,
-    // we insert `last_left_node` in `template_elements` and the seen string/template literal.
-    // ANy subsequent expression is directly inserted in `template_elements`.
+    // For each nested binary expression (with operator `+`), we keep track of the last left expressions
+    // and whether it evaluates to a string.
+    // Once we see the a string/template literal, we insert all of `left_expressions_stack` in
+    // `template_elements` and the seen string/template literal.
+    // Any subsequent expression is directly inserted in `template_elements` if its parent
+    // evaluates to a string.
     let mut template_elements = vec![];
-    let mut last_left_node = None;
+    let mut left_expressions_stack = vec![];
+    let mut binary_evaluates_to_string_stack = vec![];
+
     let mut iter = node.syntax().preorder();
     while let Some(walk) = iter.next() {
         match walk {
             WalkEvent::Enter(node) => {
-                let node = AnyJsExpression::cast(node)?;
-                match node {
+                match AnyJsExpression::cast(node)? {
                     AnyJsExpression::JsParenthesizedExpression(_) => continue,
-                    AnyJsExpression::JsBinaryExpression(node)
-                        if matches!(node.operator().ok()?, JsBinaryOperator::Plus) =>
+                    AnyJsExpression::JsBinaryExpression(ref binary)
+                        if binary.operator().ok()? == JsBinaryOperator::Plus =>
                     {
+                        left_expressions_stack.push(vec![]);
+                        binary_evaluates_to_string_stack.push(false);
                         continue;
                     }
+                    _ => {}
+                }
+                iter.skip_subtree();
```

Comment:
By moving `iter.skip_subtree()` in the match, we could remove the previous `continue` statements?
```suggestion
                    _ => iter.skip_subtree(),
                }
```